File: tests/fixtures/rust/broken_syntax.rs
//! Syntactically broken — adapter must not crash, error_count > 0.

pub struct Valid {
    pub field: i32,
}

pub fn valid_fn() -> i32 { 0 }

// Missing closing brace deliberately:
pub struct Broken {
    pub x: i32,
    pub y: i32,
// (no closing brace)

pub fn after_broken() -> i32 {
    1
}

File: tests/fixtures/cpp/broken_syntax.cpp
// Mixed valid + invalid C++ — the adapter should report parse errors
// without crashing, and still surface the well-formed parts of the
// file (here, `Healthy`) so the outline isn't empty for partially
// broken sources.

class Healthy {
public:
    void ok();
};

class Broken {
public:
    void method(  // missing closing paren
};

struct AlsoOk {
    int x;
};
